File: javascript/socketd/src/socketd/transport/client/ClientProvider.ts
import type {Client} from "./Client";
import type {ClientConfig} from "./ClientConfig";

/**
 * 客户端工厂
 *
 * @author noear
 * @since 2.0
 */
export interface ClientProvider {
    /**
     * 协议架构
     */
    schemas(): string[];

    /**
     * 创建客户端
     */
    createClient(clientConfig: ClientConfig): Client;
}

File: python/socketd/socketd/transport/client/ClientProvider.py
from .Client import Client
from socketd.transport.client.ClientConfig import ClientConfig


class ClientProvider:

    """
        协议架构
    """
    def schema(self) -> list[str]: ...

    """
        创建客户端
    """
    def create_client(self, clientConfig: ClientConfig) -> Client: ...
